### PR TITLE
feat(odyssey-react): use Text component in Field

### PR DESCRIPTION
**Pull request by andrewberg-okta**

### CODE REVIEW

File: packages/odyssey-react/src/components/Field/Field.module.scss [hunk 1]
```diff
@@ -29,27 +29,17 @@
...
   margin-block-start: -$spacing-xs;
   margin-block-end: $spacing-xs;
-  color: $text-body;
-  font-size: $size-body-caption;
 }
 
 .error {
   margin-block-start: $spacing-xs;
-  color: $text-danger;
-  font-size: $size-body-caption;
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Field/Field.module.scss:54]: \o/

File: packages/odyssey-react/src/components/Field/index.tsx [hunk 2]
```diff
@@ -10,153 +10,4 @@
...
-Field.displayName = "Field";
-Label.displayName = "FieldLabel";
-Hint.displayName = "FieldHint";
-Error.displayName = "FieldError";
-
-Field = withStyles(styles)(Field);
-
-export type FieldProps = ComponentProps<typeof Field>;
-export { Field };
+export * from "./Field";
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Field/index.tsx:13]: thanks for setting up barrel file

File: packages/odyssey-react/src/components/Text/Text.module.scss [hunk 3]
```diff
@@ -268,6 +268,10 @@
   color: $text-danger-disabled;
 }
 
+.headingColor {
+  color: $text-heading;
+}
+
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Text/Text.module.scss:274]: some duplicate work with @arnoldsandoval-okta over here: https://github.com/okta/odyssey/pull/1176/files#r749664327

Can you two please coordinate on the styles to be changed here? And if <Text /> cannot render h* tags are there better names to be using for these selectors?

File: packages/odyssey-react/src/components/Field/Field.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            optionalLabel
+          </Text>
+        </span>
+      )}
+    </Tag>
+  );
+
+  const labelVisuallyHidden = (
+    <ScreenReaderText>
+      <Tag htmlFor={inputId}>
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Field/Field.tsx:139]: this is interesting - does `<ScreenReaderText />` not support a `htmlFor` attribute here? Or is the `<Tag>` element here to allow more nuanced expression when reading this content back for example strong or em tags?

File: packages/odyssey-react/src/components/Field/Field.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      </Text>
+    </p>
+  );
+}
+
+function Error({ id, children }: PropsError) {
+  return (
+    <p className={styles.error} id={`${id}-error`}>
+      <Text color="danger" size="caption">
+        {children}
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Field/Field.tsx:163]: I notice that children is typed as `ReactNode` here (really anything) and `Hint` above has its children typed as `ReactText` - should we unify here? Keep in mind that might be a breaking change...

File: packages/odyssey-react/src/components/Field/Field.tsx (hunk 4 above)
Comment (edburyenegren-okta) [reply packages/odyssey-react/src/components/Field/Field.tsx:139]: `Tag` transitions between `label` and `legend` depending on the use case / input.

File: packages/odyssey-react/src/components/Field/Field.tsx (hunk 5 above)
Comment (edburyenegren-okta) [reply packages/odyssey-react/src/components/Field/Field.tsx:163]: `Error` requires a `ScreenReaderText` be present, so `ReactText` is insufficient.

**Review by jeffbelser-okta — APPROVED** [review]
strong work 💪 looks like VRT is unhappy though

**Pull request merged**